yellowbrick/text/postag.py

```diff
@@ -20,125 +20,371 @@
+                        self.tag_map["preposition"] += 1
+                    elif tag == "POS":
+                        self.tag_map["possessive"] += 1
+                    elif tag == "LS":
+                        self.tag_map["list"] += 1
+                    elif tag == "MD":
+                        self.tag_map["modal"] += 1
+                    elif tag in self.punct_list:
+                        self.tag_map["punctuation"] += 1
+                    elif tag == "TO":
+                        self.tag_map["infinitive"] += 1
+                    elif tag == "UH":
+                        self.tag_map["interjection"] += 1
+                    elif tag == "SYM":
+                        self.tag_map["symbol"] += 1
+                    else:
+                        self.tag_map["other"] += 1
+                
+    def draw(self, **kwargs):
         """
-        The transform method transforms the raw text input for the
-        part-of-speech tagging visualization. It requires that
-        documents be in the form of (tag, token) tuples.
+        Called from the fit method, this method creates the canvas and
+        draws the part-of-speech tag mapping as a bar chart.
 
         Parameters
         ----------
-        tagged_token_tuples : list of tuples
-            A list of (tag, token) tuples
+        kwargs: dict
+            generic keyword arguments.
+
+        Returns
+        -------
+        ax : matplotlib axes
+            Axes on which the PosTagVisualizer was drawn.
+        """
+        colors = resolve_colors(
+            n_colors=len(self.tag_map), 
+            colormap=self.colormap, 
+            colors=self.colors
+        )
+        self.ax.bar(
+            range(len(self.tag_map)), 
+            list(self.tag_map.values()), 
+            color=colors
+        )
+        return self.ax
+        
+    def finalize(self, **kwargs):
+        """
+        Finalize the plot with ticks, labels, and title
 
-        Text documents must be tokenized and tagged before passing to fit()
+        Parameters
+        ----------
+        kwargs: dict
+            generic keyword arguments.
         """
-        self.tagged = [
-            (self.TAGS.get(tag),tok) for tok, tag in tagged_tuples
-        ]
-        #
-        # print(' '.join((colorize(token, color) for color, token in self.tagged)))
-        # print('\n')
+        # NOTE: not deduping here, so this is total, not unique
+        self.set_title(
+            "PosTag plot for {}-token corpus".format(
+                (sum(self.tag_map.values()))
+            )
+        )
+        
+        self.ax.set_xticks(range(len(self.tag_map)))
+        self.ax.set_xticklabels(list(self.tag_map.keys()), rotation=90)
+        
+        # Set the axis labels
+        self.ax.set_xlabel(
+            '{} part-of-speech tags'.format(
+                self.tagset_names[self.tagset]
+            )
+        )
+        self.ax.set_ylabel('Count')
```

Comment:
In the gallery, I commented that `plt.tight_layout` might assist with the tick labels being cut off. From #613 it may be reasonable to add that call here to finalize. Happy to discuss that if you'd like.

Reply:
See my note and examples above. Essentially, `plt.tight_layout` does not seem to have the desired effect. Not sure what to do, but this does seem likely related to #613

Reply:
Ah, that's too bad that didn't work. @lwgray came up with a solution - but I don't think that it should block this PR; @lwgray would you be willing to try your idea on the gallery when we merge this in?